docusafe2-business-impl/src/main/java/de/adorsys/docusafe2/business/impl/bucketpathencryption/BucketPathEncryptionServiceImpl.java

```diff
@@ -40,37 +46,36 @@ public BucketPath encrypt(SecretKeySpec secretKey, BucketPath bucketPath) {
     }
 
     @Override
-    public BucketPath decrypt(SecretKeySpec secretKey, BucketPath bucketPath) {
-        Cipher cipher = createCipher(secretKey, Cipher.DECRYPT_MODE);
+    public BucketPath decrypt(SecretKey secretKey, BucketPath bucketPath) {
+        Optional<Cipher> cipher = createCipher(secretKey, Cipher.DECRYPT_MODE);
+
+        cipher.orElseThrow(() -> new BaseException("Can't build cipher, secret key is absent"));
 
         List<String> subdirs = BucketPathUtil.split(BucketPathUtil.getAsString(bucketPath));
         StringBuilder decryptedPathString = new StringBuilder();
         for(String subdir : subdirs) {
             byte[] decrypt = HexUtil.convertHexStringToBytes(subdir.toUpperCase());
             byte[] decryptedBytes = new byte[0];
             try {
-                decryptedBytes = cipher.doFinal(decrypt);
+                decryptedBytes = cipher.get().doFinal(decrypt);
             } catch (IllegalBlockSizeException | BadPaddingException e) {
-                e.printStackTrace();
+                log.error("Error during decryption", e.getMessage());
             }
             decryptedPathString.append(BucketPath.BUCKET_SEPARATOR).append(new String(decryptedBytes, UTF_8));
         }
         return new BucketPath(decryptedPathString.toString());
     }
 
-    private static Cipher createCipher(SecretKeySpec secretKey, int cipherMode) {
+    private Optional<Cipher> createCipher(SecretKey secretKey, int cipherMode) {
```

Comment:
For me it wrapping into optional makes no sense, if secretKey is null we can throw immediately here:
```
if(secretKey == null) return Optional.empty();
```

Reply:
I agree with valentyn.